### PR TITLE
Delete deprecated versions of get*AuthDataGenerator that don't use serviceMode.

### PR DESCRIPTION
**Pull request by seehamrun**
For #25

### CODE REVIEW

File: portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java [hunk 1]
```diff
@@ -46,37 +46,19 @@
      * have permissions based on the serviceMode provided.
      */
     default OfflineAuthDataGenerator getOfflineAuthDataGenerator(PortableDataType dataType, ServiceMode serviceMode) {
-      // default implementation to ignore serviceMode
-      return getOfflineAuthDataGenerator(dataType);
+      logger.debug("WARNING: getOnlineAuthDataGenerator not implemented for type: {}, serviceMode: {}, service: {}",
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java]: how about logger.warn( ?

Comment (seehamrun) [reply portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java]: Done.

File: portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java [hunk 2]
```diff
@@ -46,37 +46,19 @@
      * have permissions based on the serviceMode provided.
      */
     default OfflineAuthDataGenerator getOfflineAuthDataGenerator(PortableDataType dataType, ServiceMode serviceMode) {
-      // default implementation to ignore serviceMode
-      return getOfflineAuthDataGenerator(dataType);
+      logger.warn("WARNING: getOnlineAuthDataGenerator not implemented for type: {}, serviceMode: {}, service: {}",
```
Comment (holachuy) [inline portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java]: One last thing, you can remove the WARNING: from the log statement as the warn() method will prepend that to the output. Here and below.

Comment (seehamrun) [reply portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java]: Good catch! Done.

**Pull request merged**